src/app/clusters/camera-av-settings-user-level-management-server/camera-av-settings-user-level-management-server.cpp

```diff
@@ -0,0 +1,1079 @@
+    }
+
+    if (SupportsOptAttr(OptionalAttributes::kTiltMax))
+    {
+        VerifyOrReturnError(HasFeature(Feature::kMechanicalTilt), CHIP_ERROR_INVALID_ARGUMENT,
+                            ChipLogError(Zcl,
+                                         "CameraAVSettingsUserLevelManagement[ep=%d]: Feature configuration error. If TiltMax is "
+                                         "enabled, then MechanicalTilt feature is required",
+                                         mEndpointId));
+    }
+
+    if (SupportsOptAttr(OptionalAttributes::kPanMin))
+    {
+        VerifyOrReturnError(HasFeature(Feature::kMechanicalPan), CHIP_ERROR_INVALID_ARGUMENT,
+                            ChipLogError(Zcl,
+                                         "CameraAVSettingsUserLevelManagement[ep=%d]: Feature configuration error. If PanMin is "
+                                         "enabled, then MechanicalPan feature is required",
+                                         mEndpointId));
+    }
+
+    if (SupportsOptAttr(OptionalAttributes::kPanMax))
+    {
+        VerifyOrReturnError(HasFeature(Feature::kMechanicalPan), CHIP_ERROR_INVALID_ARGUMENT,
+                            ChipLogError(Zcl,
+                                         "CameraAVSettingsUserLevelManagement[ep=%d]: Feature configuration error. If PanMax is "
+                                         "enabled, then MechanicalPan feature is required",
+                                         mEndpointId));
+    }
+
+    // Set default MPTZ
+    mMptzPosition.pan  = Optional<int16_t>(defaultPan);
+    mMptzPosition.tilt = Optional<int16_t>(defaultTilt);
+    mMptzPosition.zoom = Optional<uint8_t>(defaultZoom);
+
+    VerifyOrReturnError(AttributeAccessInterfaceRegistry::Instance().Register(this), CHIP_ERROR_INTERNAL);
+    ReturnErrorOnFailure(CommandHandlerInterfaceRegistry::Instance().RegisterCommandHandler(this));
+    return CHIP_NO_ERROR;
+}
+
+bool CameraAvSettingsUserLevelMgmtServer::HasFeature(Feature aFeature) const
+{
+    return mFeatures.Has(aFeature);
+}
+
+bool CameraAvSettingsUserLevelMgmtServer::SupportsOptAttr(OptionalAttributes aOptionalAttr) const
+{
+    return mOptionalAttrs.Has(aOptionalAttr);
+}
+
+void CameraAvSettingsUserLevelMgmtServer::MarkDirty(AttributeId aAttributeId)
+{
+    MatterReportingAttributeChangeCallback(mEndpointId, CameraAvSettingsUserLevelManagement::Id, aAttributeId);
+}
+
+// Attribute mutators
+//
+CHIP_ERROR CameraAvSettingsUserLevelMgmtServer::SetMaxPresets(uint8_t aMaxPresets)
+{
+    mMaxPresets = aMaxPresets;
+    MarkDirty(Attributes::MaxPresets::Id);
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR CameraAvSettingsUserLevelMgmtServer::SetTiltMin(int16_t aTiltMin)
+{
+    if ((aTiltMin < kMinTiltValue) || (aTiltMin > kMaxTiltValue - 1) || (aTiltMin > mTiltMax))
+    {
+        return CHIP_IM_GLOBAL_STATUS(ConstraintError);
+    }
+
+    mTiltMin = aTiltMin;
+    MarkDirty(Attributes::TiltMin::Id);
```

Comment:
Should not MarkDirty if the value did not actually change..... That applies to all the setters.

Reply:
Fixed in this, and all the other set methods